src/libcore/mem.rs

```diff
@@ -611,3 +611,38 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
```

Comment:
> This is for C-like (i.e. numeric/data-less) enums only.

The reference does not restrict or qualify this statement in any way. If you want to remove this as a guarantee for all enums, then change the reference. I do understand the reluctance to make guarantees, but this is a bit ridiculous. What would the motivation be for changing discriminants according to the phase of the moon?